### PR TITLE
Corner-case coverage and hardening for the 1.0 release

### PR DESCRIPTION
**Pull request by octaviospain**
Closes #201

## Summary

Hardens the library against corner cases and locks down documented behavior ahead of the 1.0 release, backed by a broad new test matrix. Verified end to end; the default test suite is green (878 tests, 0 failures) and the opt-in concurrency stress test passes.

### Source hardening
- **Physical-identity `uniqueId`** — now `fileName` + duration (whole seconds) + bit rate, dropping the mutable title term, so re-tagging a file keeps its identity stable.
- **Bounded `newId()`** — the audio-library and playlist bases fail fast both when the id space is densely populated and when the `AtomicInteger` counter overflows `Int.MAX_VALUE` (previously it silently handed out negative ids).
- **Foreign-origin entities** log a warning and are re-wired on add.
- **Album cover reload** — the no-cover sentinel resets when a `SoftReference` cover is GC-evicted, so a later access can reload instead of returning `null` permanently.
- **Metadata-IO init window** closed structurally, before any catalog subscription can observe items.
- **Playlist guards** — `createPlaylist` rejects the unassigned id, and the audio-item delete subscriber is made non-bypassable via a first-use guard applied to every playlist mutation entry point.
- **Duplicate file-path policy** — adding the same physical file twice returns the existing item (dedup by `uniqueId`).
- **Per-library instance name** — an optional builder setter with an auto-generated stable default, surfaced as a `libraryInstance` logging MDC key across library operations and import-service scopes.

### Tests
- Cross-library operation matrix (Core + FX) over baseline, foreign-object, and sequential-lifecycle axes, exercised through the reactive base modules and the opt-in persistence-mapping modules.
- Subscriber-isolation test: a throwing `onNext` does not cancel peer subscribers.
- JSON round-trip across all three persistence-mapping modules plus a SQLite smoke test; a degenerate/empty-input sweep; regression tests pinning the source fixes and the post-close refresh no-op.
- Quarantined concurrent-import stress test (opt-in via `-PincludeStress=true`) asserting no lost or duplicate writes and no leaked scopes over a 10k-item in-memory corpus.

### Housekeeping
- Added the missing GPL license header to the iTunes import service.

## Testing

- `gradle compileKotlin compileTestKotlin ktlintCheck test` — green (878 tests, 0 failures).
- `gradle :music-commons-core:test -PincludeStress=true --tests "*ConcurrentImportStressTest*"` — passes (final repository size 10000, no duplicate ids, no leaked library reference).

## Notes

- The `uniqueId` string format changed (title term removed). Persistence keys entities by their integer repository id, not by `uniqueId`, so no stored-data migration is required; only consumers that parse or persist the raw `uniqueId` string are affected. Breaking changes to the public API are permitted in this pre-1.0 line.

### CODE REVIEW

File: music-commons-core/src/main/kotlin/net/transgressoft/commons/music/CoreMusicLibrary.kt [hunk 1]
```diff
@@ -274,6 +300,8 @@ class CoreMusicLibrary private constructor(
          * Rollback on partial failure closes everything constructed so far before rethrowing.
          */
         fun build(): CoreMusicLibrary {
+            val resolvedName = instanceName ?: "music-library-${instanceCounter.incrementAndGet()}"
+
```
[the remark this answers is not in the document]
Comment (octaviospain) [reply music-commons-core/src/main/kotlin/net/transgressoft/commons/music/CoreMusicLibrary.kt]: Fixed — the auto-generated defaults now use type-distinguishing prefixes: `core-music-library-N` for `CoreMusicLibrary` and `fx-music-library-N` for `FXMusicLibrary`, so they no longer collide when both library types coexist in one JVM.

File: music-commons-core/src/main/kotlin/net/transgressoft/commons/music/playlist/PlaylistHierarchyBase.kt [hunk 2]
```diff
@@ -96,14 +155,22 @@ abstract class PlaylistHierarchyBase<I : ReactiveAudioItem<I>, P : ReactiveAudio
...
+            check(attempts++ < MAX_ID_SEARCH_ATTEMPTS) {
+                "Playlist id space exhausted: no available id found after $MAX_ID_SEARCH_ATTEMPTS attempts"
+            }
             id = idCounter.getAndIncrement()
+            check(id > 0) {
+                "Playlist id space exhausted: the id counter overflowed Int.MAX_VALUE"
+            }
         } while (contains(id))
         return id
     }
```
[the remark this answers is not in the document]
Comment (octaviospain) [reply music-commons-core/src/main/kotlin/net/transgressoft/commons/music/playlist/PlaylistHierarchyBase.kt:175]: Fixed — `PlaylistHierarchyBase` now seeds its id counter lazily to `max(existing id) + 1` on first allocation, so allocation stays O(1) on a hierarchy loaded with existing playlists and the retry cap only guards a genuinely saturated id space. The audio-library base keeps its unseeded counter but with a 1,000,000-attempt cap plus a new overflow guard; the dense-low-id scenario is not realistically reachable there, but I can add the same seeding for symmetry if preferred.

File: music-commons-core/src/test/kotlin/net/transgressoft/commons/music/CoreMusicLibraryBuilderInstanceNameTest.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        val first = CoreMusicLibrary.builder().build()
+        first.close()
+        val second = CoreMusicLibrary.builder().build()
+        try {
+            first.instanceName shouldNotBe second.instanceName
+        } finally {
+            second.close()
+        }
+    }
+})
```
[the remark this answers is not in the document]
Comment (octaviospain) [reply music-commons-core/src/test/kotlin/net/transgressoft/commons/music/CoreMusicLibraryBuilderInstanceNameTest.kt:49]: Added — `CoreMusicLibraryBuilderInstanceNameTest` now asserts that both `instanceName("")` and `instanceName("   ")` throw `IllegalArgumentException`.

**Comment by octaviospain** [comment]
Addressed the two outside-diff-range items: `createPlaylistDirectory(name, audioItems)` now applies the same `UNASSIGNED_ID` guard as the `createPlaylist` overload in both `DefaultPlaylistHierarchy` and `FXPlaylistHierarchy`, rejecting unassigned (zero) ids before the directory is constructed so unpersisted audio items cannot create dangling references.
